src/flag_gems/ops/masked_fill.py

```diff
@@ -4,35 +4,16 @@
 import triton
 import triton.language as tl
 
-from .. import runtime
-from ..utils import broadcastable_to, libentry
-from ..utils import triton_lang_extension as tle
+from ..utils import broadcastable_to, pointwise_dynamic
 
 
-@libentry()
-@triton.autotune(configs=runtime.get_tuned_config("masked_fill"), key=["N"])
+@pointwise_dynamic(
+    is_tensor=[True, True, False], promotion_methods=[(0, 1, 2, "NO_OPMATH")]
+)

```

Comment:
The output type should be `[(0, "NO_OPMATH")]`.
Please refer to `docs/pointwise_dynamic.md` for more details. Or contact us.